src/map_preview_graphic.c

```diff
@@ -0,0 +1,946 @@
+        SetBgControlParam(unk->bgConfig, 2, GF_BG_CNT_SET_COLOR_MODE, 1);
+        SetBgPriority(GF_BG_LYR_MAIN_3, 0);
+        SetBgPriority(GF_BG_LYR_MAIN_2, 1);
+        SetBgPriority(GF_BG_LYR_MAIN_0, 2);
+        MapPreviewGraphic_LoadGfx(unk);
+        MapPreviewGraphic_InitWindow(unk);
+        G2_SetBlendAlpha(4, 33, 16, 0);
+        BeginNormalPaletteFade(0, 1, 1, RGB_BLACK, 16, 1, unk->heapId);
+        unk->state++;
+        break;
+    case PG_STATE_FADE_IN:
+        if (IsPaletteFadeFinished()) {
+            unk->state++;
+        }
+        break;
+    case PG_STATE_WAIT_OR_SKIP:
+        if (gSystem.newKeys & PAD_BUTTON_A) {
+            unk->fadeOutTimer = 0;
+            unk->state++;
+            break;
+        }
+        if (++unk->displayTimer > 60) {
+            unk->fadeOutTimer = 0;
+            unk->state++;
+        }
+        break;
+    case PG_STATE_FADE_OUT:
+        if (unk->fadeOutTimer < 16) {
+            if (++unk->unk1A >= 2) {
+                unk->unk1A = 0;
+                unk->fadeOutTimer++;
+            } 
+            reg_G2_BLDALPHA = (16 - unk->fadeOutTimer) | (unk->fadeOutTimer << 8);
+        } 
+        if (sub_0206A694(unk, fsys) && unk->fadeOutTimer >= 16) {
+            ToggleBgLayer(2, GF_PLANE_TOGGLE_OFF);
+            SetBgControlParam(unk->bgConfig, 2, GF_BG_CNT_SET_COLOR_MODE, unk->bgColorMode);
+            SetBgPriority(GF_BG_LYR_MAIN_2, (u8) unk->bg2Priority);
+            SetBgPriority(GF_BG_LYR_MAIN_0, (u8) unk->bg0Priority);
+            SetBgPriority(GF_BG_LYR_MAIN_1, (u8) unk->bg1Priority);
+            SetBgPriority(GF_BG_LYR_MAIN_3, (u8) unk->bg3Priority);
+            reg_G2_BLDCNT = 0;
+            BG_ClearCharDataRange(2, 0x20, 0, unk->heapId);
+            BgClearTilemapBufferAndCommit(unk->bgConfig, 2);
+            WindowArray_Delete(unk->window, 1);
+            FreeToHeap(unk);
+            return TRUE;
+        }
+        break;
+    }
+    return FALSE;
+}
+
+static void MapPreviewGraphic_LoadGfx(UnkStruct_0206A388 *unk) {
+    GfGfxLoader_GXLoadPal(NARC_a_1_5_0, sMapPreviewGraphicData[unk->index].files[unk->timeIndex].palNo, GF_PAL_LOCATION_MAIN_BG, GF_PAL_SLOT_0_OFFSET, 0x160, unk->heapId);
+    GfGfxLoader_LoadCharData(NARC_a_1_5_0, sMapPreviewGraphicData[unk->index].files[unk->timeIndex].charNo, unk->bgConfig, GF_BG_LYR_MAIN_2, 0, 0, 1, unk->heapId);
+    GfGfxLoader_LoadScrnData(NARC_a_1_5_0, sMapPreviewGraphicData[unk->index].files[unk->timeIndex].scrnNo, unk->bgConfig, GF_BG_LYR_MAIN_2, 0, 0, 1, unk->heapId);
+}
+
+static void MapPreviewGraphic_InitWindow(UnkStruct_0206A388 *unk) {
+    unk->window = AllocWindows(unk->heapId, 1);
+    AddWindowParameterized(unk->bgConfig, unk->window, 2, 0, 0, 28, 2, 13, 0x2C0);
+}
+
+static int sub_0206A694(UnkStruct_0206A388 *unk, FieldSystem *fsys) {
+    LocalMapObject *player;
+    
+    if (!unk->unk30) {
+        return TRUE;
+    }
+
+    UnkStateMachineSubstruct_0206A388 *unkSub = &unk->unk34; //required to match
```

Comment:
This comment seems kinda redundant

Reply:
it was just a really weird thing to be necessary tbh